### ISSUE TITLE
Ctrl + Shift + Down (Swap Line Down) fails to swap to an empty last line

### ISSUE THREAD

**Issue by titoBouzout** [SublimeTextIssues/DefaultPackages#86]
_From @ThoAppelsin on February 12, 2014 22:55_

In an example case of:

>1 This is the first line
>2 Example
>3
>4 Random
>5 Text
>6

While having the caret at the line `#5`, the key combination for (Swap Line Down) or doing it through the menu, fails to swap line `#6` and line `#5`, and to make the "Text" be at the last line.

This happens only when the last line is empty.

One additional thing is that the last line in this case (in the case when the last line is empty), cannot be swapped above; so having the caret at line `#6` and making a (Swap Line Up) also doesn't work.

This is for Sublime Text 3.

Thanks for reading...

_Copied from original issue: SublimeTextIssues/Core#244_

**Comment by titoBouzout** [comment]
_From @FichteFoll on February 13, 2014 0:55_

Can reproduce on 3061. I guess this is intended behavior since the last line is essentially treated as the end of the file by many (ST even has an option to enforce that when saving, `ensure_newline_at_eof_on_save`; Python PEP8 also recommends this).
Setting that option to `false` doesn't make it work btw.

**Comment by titoBouzout** [comment]
_From @FichteFoll on December 22, 2014 15:57_

Imo, this is still intended behavior and thus not a bug. The newline at EOF is in fact treaded as the EOF and results in the same functionality as if there was no EOF: no swapping takes place.

So, this is`R: invalid` to me. Any other comments?

**Comment by titoBouzout** [comment]
I consider it a bug, as if I don't want to move the line there, just don't move it. If I'm trying but the line does not move, then its an unwanted limitation.

**Comment by titoBouzout** [comment]
_From @FichteFoll on December 22, 2014 17:22_

Please elaborate, I don't understand.

**Comment by titoBouzout** [comment]
I've 

>1 This is the first line
>2 Example
>3
>4 Random
>5 Text
>6

I want to swap to the following but I can't

>1 This is the first line
>2 Example
>3
>4 Random
>5 
>6 Text

**Comment by titoBouzout** [comment]
_From @FichteFoll on December 22, 2014 22:57_

How do you know whether it's an "unwanted limitation" or not? I checked the source code and it does not seem to be some bug that slipped through.

But either way, only Jon can truly say whether this is intended or not.

**Comment by titoBouzout** [comment]
Forcing a newline at EOF is not good. I want to swap the line, the one that reported the bug too.

**Comment by titoBouzout** [comment]
_From @FichteFoll on December 22, 2014 23:18_

I want a newline at EOF, it is in fact the "new EOF" and I would also not swap. Since there are some people (Jon probably included) that prefer the EOF way this should be an option or something, in which case it would qualify as an enhancement.

**Comment by titoBouzout** [comment]
But FichteFoll, is some languages and situations is not an option, is not a
matter of personal choice. Think for example about some php files that are
included before sending any http header. A new line will break it... I
don't consider that important if it is a bug or an enhancement, what is
important is that this shouldn't be forcing it.

**Comment by titoBouzout** [comment]
_From @ThoAppelsin on December 23, 2014 9:43_

I confirm that I wanted to do that back then. I was using the Sublime Text 3 as a simple text editor back then on that day, was hoping to put something on the last line, and at the same time I happened to desire a new-line right behind it. I recognized that a line-swap would do it, so I gave it a shot with no avail.
It might have been thought as a feature, but the implementation is a restrictive one, which is why I can safely say that it is an improper implementation. This feature, simply should have been offered along with an option to disable it, so that it doesn't cripple Sublime Text 3 like this.
